src/app/api/imageInput/route.ts

```diff
@@ -0,0 +1,158 @@
+    messages: zodMessageObject.data.message as Message[],
+  });
+
+  // console.log("zodMessage:", zodMessage);
+
+  const {
+    imageName,
+    imageType,
+    imageSize,
+    value,
+    userId,
+    orgId,
+    chatId,
+    id,
+    messages,
+    imageFile,
+  } = zodMessage;
+  const { orgSlug } = await auth();
+
+  const _message = messages as unknown as Message[];
+  // console.log("_message", _message);
+  const url = request.url;
+  const urlArray = url.split("/");
+
+  if (!formData || !formData.has("file")) {
+    return NextResponse.json(
+      { result: "no data received", success: false },
+      { status: 400 },
+    );
+  }
+  const parts = imageFile.name.split(".");
+  const extension = parts[parts.length - 1];
+  // console.log("messages",messages)
+  const formattedMessages = messages.map((msg: any) => {
+    return {
+      role: msg.role,
+      content: msg.content,
+    };
+  });
+  // console.log("formatemsgs", formattedMessages);
+  const combinedContent = formattedMessages
+    .map((msg: any) => msg.content)
+    .join("\n");
+  const combinedMessage = {
+    role: "user",
+    content: combinedContent,
+  };
+  console.log("Combined message:", combinedMessage.content);
+
+  if (file && zodMessage && combinedMessage) {
+    const blob = file as Blob;
+    const buffer = Buffer.from(await blob.arrayBuffer());
+    const imageBase64 = buffer.toString("base64");
+    const message = new HumanMessage({
+      content: [
+        {
+          type: "text",
+          text: combinedMessage.content,
+        },
+        {
+          type: "image_url",
+          image_url: `data:image/png;base64,${imageBase64}`,
+        },
+      ],
+    });
+    const res = await Promise.race([
+      chat.invoke([message]),
+      new Promise((resolve, reject) => {
+        setTimeout(() => reject(new Error("Timeout occurred")), 50000);
+      }),
+    ]).catch((error) => {
+      console.error("Error:", error);
+      return null; // Handle the timeout error appropriately
+    });
+    console.log("Image response", res);
+    if (res) {
+      const latestReponse = {
+        id: id,
+        role: "assistant" as const,
+        content: typeof res === "string" ? res : JSON.stringify(res),
+        createdAt: new Date(),
+        audio: "",
+      };
+      const db = await saveToDB({
+        _chat: _message,
+        chatId: chatId,
+        orgSlug: orgSlug as string,
+        latestResponse: latestReponse,
+        userId: userId,
+        orgId: orgId,
+        urlArray: urlArray,
+      });
+      console.log("dbResponce", db);
+
+      const saveDroppedImag = await saveDroppedImage({
+        imageId: zodMessageObject.data.id,
+        buffer: buffer,
+        chatId: zodMessageObject.data.chatId,
+        imageExtension: extension,
+        imageName: file.name,
+      });
+
+      return NextResponse.json(
+        { result: res, imageUrl: saveDroppedImag, success: true },
+        { status: 200 },
+      );
+    }
```

Comment:
can't we do proper streaming instead of faking it on frontend

Reply:
older one